### PR TITLE
infra: allow ssh into nodes to make debugging easier

### PR DESCRIPTION
**Pull request by yangchiu**
infra: allow ssh into nodes to make debugging easier

For https://github.com/longhorn/longhorn/issues/6601

Signed-off-by: Yang Chiu <yang.chiu@suse.com>

### CODE REVIEW

File: pipelines/utilities/kubeconfig.sh [hunk 1]
```diff
@@ -12,6 +12,8 @@ set_kubeconfig(){
   elif [[ "${TF_VAR_arch}" == "arm64"  ]]; then
     export KUBECONFIG="${PWD}/test_framework/terraform/${LONGHORN_TEST_CLOUDPROVIDER}/${DISTRO}/k3s.yaml"
   fi
+
+  cat "${KUBECONFIG}"
```
Comment (innobead) [inline pipelines/utilities/kubeconfig.sh:16]: Could this expose credentials outside in the job build?

**Review by innobead — CHANGES_REQUESTED** [review]
Suggest making exporting worker nodes public optional, so this could make us have more flexibilty for testing env setup.

File: pipelines/utilities/kubeconfig.sh (hunk 1 above)
Comment (yangchiu) [reply pipelines/utilities/kubeconfig.sh:16]: From my understanding there are 3 fields in KUBECONFIG: `certificate-authority-data`, `client-certificate-data` and `client-key-data`. `certificate-authority-data` is SSL server certificate. `client-certificate-data` and `client-key-data` are also `openssl` generated certificate and key which to be approved to access the k8s cluster, so they don't involve personal ssh key or credential. Even though one gets this KUBECONFIG file, he can only manipulate the created k8s cluster / ec2 instances, and has nothing to do with our infra machines.

**Comment by yangchiu** [comment]
> Suggest making exporting worker nodes public optional, so this could make us have more flexibilty for testing env setup.

I'll add this option.

**Comment by yangchiu** [comment]
> > Suggest making exporting worker nodes public optional, so this could make us have more flexibilty for testing env setup.
> 
> I'll add this option.

Switching between public ip and private network is more complicated than expected. Simply disabled `associate_public_ip_address` won't work. Need more time to investigate.

**Comment by yangchiu** [comment]
> Switching between public ip and private network is more complicated than expected. Simply disabled `associate_public_ip_address` won't work. Need more time to investigate.

Refined. If it has a public ip, it can directly communicate with Internet. If it doesn't have a public ip, use nat to communicate with Internet.

**Comment by yangchiu** [comment]
> Suggest making exporting worker nodes public optional, so this could make us have more flexibilty for testing env setup.

Added option in https://github.com/longhorn/infra/pull/153.

File: pipelines/utilities/kubeconfig.sh (hunk 1 above)
Comment (yangchiu) [reply pipelines/utilities/kubeconfig.sh]: We don't directly print out `kubeconfig` now, but users can ssh into control plane node to retrieve it. For example, the default file path for `k3s` is `/etc/rancher/k3s/k3s.yaml`.

**Review by innobead — CHANGES_REQUESTED** [review]
Instead of removing the private network for worker nodes, have you consided bastion way to ssh from local to the internal nodes?

Suggest this way to avoid avoidable instance access exposure (security). 
https://repost.aws/knowledge-center/ec2-troubleshoot-bastion-host-connection

**Comment by yangchiu** [comment]
More investigation needed. Close this for now.

**Comment by innobead** [comment]
Sounds good @yangchiu

**Pull request closed without merging**